docs/design/baremetal/networking-infrastructure.md

```diff
@@ -70,7 +70,101 @@ DNS requirements internal to the cluster as possible, leaving only a
 small amount of public DNS configuration to be implemented by the user
 before starting the installation process.
 
-TODO - explain how this works ...
+In a `baremetal` environment, we do not know the IP addresses of all hosts in
+advance.  Those will come from an organization’s DHCP server.  Further, we can
+not rely on being able to program an organization’s DNS infrastructure in all
+cases.  We address these challenges in two ways:
 
-Explain that $cluster_name and $base_domain are supplied via
-install-config.yaml
+1. Self host some DNS infrastructure to provide DNS resolution for records only
+   needed internal to the cluster.
+2. Make use of mDNS (Multicast DNS) to dynamically discover the addresses of
+   hosts that we must resolve records for.
+
+### mdns-publisher
+
+https://github.com/openshift/mdns-publisher
+
+The `mdns-publisher` is the component that runs on each host to make itself
+discoverable by other hosts in the cluster.  It will be run by the
+`machine-config-operator`, once the following PR is complete:
+
+https://github.com/openshift/machine-config-operator/pull/795
+
+### coredns-mdns
+
+https://github.com/openshift/coredns-mdns/
+
+The `mdns` plugin for `coredns` was developed to perform DNS lookups
+based on discoverable information from mDNS.  This plugin will resolve both the
+`etcd-NNN` records, as well as the `_etcd-server-ssl._tcp.` SRV record.
+
+The list of `etcd` hosts included in the SRV record is based on the list of
+control plane nodes currently running.
+
+The IP addresses that the `etcd-NNN` host records resolve to comes from the
+mDNS advertisement sent out by the `mdns-publisher` on that control plane noded
```

Comment:
Bit of a typo - "that control plane noded"

Reply:
fixed